apricot/local/servicehttp.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2019-2021 CERN and copyright holders of ALICE O².
+ * Author: Claire Guyot <claire.eloise.guyot@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package local
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/AliceO2Group/Control/configuration"
+	"github.com/gorilla/mux"
+)
+
+type HttpService struct {
+	svc configuration.Service
+}
+
+func (httpsvc *HttpService) ApiGetFlps(w http.ResponseWriter, r *http.Request) {
+	keyPrefix := "o2/hardware/flps/"
+	httpsvc.ApiGetClusterInformation(w, r, keyPrefix)
+}
+
+func (httpsvc *HttpService) ApiGetDetectorFlps(w http.ResponseWriter, r *http.Request) {
+	keyPrefix := "o2/hardware/detectors/TST/flps/"
+	httpsvc.ApiGetClusterInformation(w, r, keyPrefix)
+}
+
+func (httpsvc *HttpService) ApiGetClusterInformation(w http.ResponseWriter, r *http.Request, keyPrefix string) {
+	queryParam := mux.Vars(r)
+	format := ""
+	format = queryParam["format"]
+	if format == "" {
+		format = "text"
+	}
+	keys, err := httpsvc.svc.GetHostInventory(keyPrefix)
+	if err != nil {
+		log.WithError(err).Fatal("Error, could not retrieve host list.")
```

Comment:
Not really a `Fatal` IMO, `Fatal` should only be used if we're forced to shut down the instance or the instance is already dying/crashing. For me this is an honest `Warn`.